api/src/main/java/org/eclipse/microprofile/reactive/messaging/connector/OutgoingConnectorFactory.java

```diff
@@ -0,0 +1,108 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.reactive.messaging.connector;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.MessagingProvider;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+
+import java.util.NoSuchElementException;
+
+/**
+ * SPI used to implement a connector managing a sink of messages for a specific <em>transport</em>. Typically, to
+ * handle dispatch messages to Kafka, the reactive messaging extension would need to implement a {@code bean}
+ * implementing this interface. This bean is called for every {@code stream} that need to be created for this specific
+ * <em>transport</em> (so Kafka in the example). These streams are connected to methods annotated with
+ * {@link org.eclipse.microprofile.reactive.messaging.Outgoing}.
+ * <p>
+ * The factory is called to create a {@code subscriber} for each configured <em>transport</em>. The configuration is done
+ * using MicroProfile Config. The following snippet gives an example for a hypothetical Kafka connector:
+ *
+ * <pre>
+ * mp.messaging.outgoing.my-stream.type=i.e.m.reactive.messaging.impl.kafka.Kafka
+ * mp.messaging.outgoing.my-stream.bootstrap-servers=localhost:9092
+ * mp.messaging.outgoing.my-stream.topic=my-topic
+ * ...
+ * </pre>
+ * <p>
+ * This configuration keys are structured as follows: {@code mp.messaging.[incoming|outgoing].stream-name.attribute}.
+ * <p>
+ * The {@code stream-name} segment in the configuration key corresponds to the name of the stream used in the
+ * {@code Incoming} annotation:
+ *
+ * <pre>
+ * &#64;Outgoing("my-stream")
+ * public CompletionStage&lt;String&gt; produce(String s) {
+ *      // ...
+ * }
+ * </pre>
+ * <p>
+ * The set of attributes depend on the connector and transport layer (typically, bootstrap-servers is Kafka specific). The
+ * {@code type} attribute is mandatory and indicates the fully qualified name of the {@link MessagingProvider}
+ * implementation. It must match the class returned by the {@link #type()} method. This is how a reactive messaging
+ * implementation looks for the specific {@link OutgoingConnectorFactory} required for a stream. In the previous
+ * configuration, the reactive extension implementation would need to find the {@link OutgoingConnectorFactory} returning
+ * the {@code i.e.m.reactive.messaging.impl.kafka.Kafka} class as result to its {@link #type()} method to create the
+ * {@code my-stream} <em>subscriber</em>. Note that if the connector cannot be found, the deployment must be failed.
+ * <p>
+ * The {@link #getSubscriberBuilder(Config)} is called for every <em>sink</em> that needs to be created. The
+ * {@link Config} object passed to the method contains a subset of the global configuration, and with the prefixes removed.
```

Comment:
For the production of thus subset I would like to discuss a filter that can:
wildcarding for the stream-name
wildcard/either for incoming/outgoing

Reply:
as the stream-name would likely be different, I don't see how you would do it. NEed to be discussed during the hangout.

Reply:
i.e. set a property for all streams that use a particular provider. So for example if there is a property that is the same for all streams that use Kafka.
It seems to me that these properties will not be particularly easy to set up when there are lots of streams
and many of the streams will have identical properties that relate to the install or the users principle etc.

Reply:
Then, they are in the connector bean configuration, not in the channel configuration. 
As the channel names are going to be different, what would be the prefix? 

`mp.messaging.[???]`

I don't really see the point as the connector can simply use:
```
@ConfigProperty(name="boostrap.servers") String servers;
```

And then the channels don't have to indicate the server anymore (only if they want another broker)

Reply:
There is Servicability benefits if we trace the Config object we pass in and it contains what was used rather than encourage an approach that will make less likely Connector imples behave the same way? To take this further would be to allow every needed Connector property to be done in a custom way.